### PR TITLE
fix(components): [input] textarea height calc

### LINKED ISSUE #12074: [Component] [input] el-input attribute: autosize is not enabled when I change the  visibility of component by v-show
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.3.0`
- Browser / OS: `Chrome/111.0.0.0 Safari/537.36 Edg/111.0.1661.41`
- Build Tool: `Vite`

### Related Component

- `el-input`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IExpc3QgZnJvbSBcIi4vTGlzdC52dWVcIlxuXG5jb25zdCBzaG93VHlwZSA9IHJlZignMScpXG5jb25zdCBvbkNoYW5nZSA9ICgpID0+IHtcbiAgc2hvd1R5cGUudmFsdWUgPSBzaG93VHlwZS52YWx1ZSA9PT0gXCIxXCIgPyBcIjJcIiA6IFwiMVwiXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2PlxuICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBAY2xpY2s9XCJvbkNoYW5nZVwiPmNoYW5nZTwvZWwtYnV0dG9uPlxuICAgIDxMaXN0IHYtc2hvdz1cInNob3dUeXBlID09PSAnMSdcIiB0eXBlPVwiMVwiPjwvTGlzdD5cbiAgICA8TGlzdCB2LXNob3c9XCJzaG93VHlwZSA9PT0gJzInXCIgdHlwZT1cIjJcIj48L0xpc3Q+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiTGlzdC52dWUiOiI8c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmLCB2ZXJzaW9uIGFzIHZ1ZVZlcnNpb24gfSBmcm9tICd2dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIEVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG4gIFxuZGVmaW5lUHJvcHM8e1xuICB0eXBlOiBzdHJpbmdcbn0+KClcblxuY29uc3QgdmFsdWUgPSByZWYoXCJhSE9JRlZTREdHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dzZGdob3Nkamdwc2pncm9qam9lZWpnR0dHR0dHR0dzZGdzZXNnZWVHR0dhc2ZnZWdlZ2VzZ2dnZ2dnZ3NkYXNnc2RncmdyRS1VV1ItXCIpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2IHN0eWxlPVwid2lkdGg6IDMwMHB4XCI+XG4gICAgPGRpdj5cbiAgICAgIHt7IHR5cGUgfX1cbiAgICA8L2Rpdj5cbiAgICAgIDxlbC1pbnB1dCA6bW9kZWwtdmFsdWU9J3ZhbHVlJyByZWFkb25seSB0eXBlPVwidGV4dGFyZWFcIiA6YXV0b3NpemU9XCJ7IG1pblJvd3M6IDIsIG1heFJvd3M6IDYgfVwiPjwvZWwtaW5wdXQ+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbiIsIl9vIjp7fX0=)

### Steps to reproduce

1. click the 'change' button
see the 2nd 'List' component is changed from invisible to visible, but the `el-input` component's height doesn't auto changed.(I have set the attribute autosize: { minRows: 2, maxRows: 6 } )

```vue
// App Component
<script setup lang="ts">
import { ref, version as vueVersion } from 'vue'
import List from "./List.vue"

const showType = ref('1')
const onChange = () => {
  showType.value = showType.value === "1" ? "2" : "1"
}
</script>

<template>
  <div>
    <el-button type="primary" @click="onChange">change</el-button>
    <List v-show="showType === '1'" type="1"></List>
    <List v-show="showType === '2'" type="2"></List>
  </div>
</template>
```

```vue
// List component
<script setup lang="ts">
import { ref, version as vueVersion } from 'vue'
import { version as EpVersion } from 'element-plus'
import { ElementPlus } from '@element-plus/icons-vue'
  
defineProps<{
  type: string
}>()

const value = ref("aHOIFVSDGGGGGGGGGGGGGGGGGGGGGGGGGGGGsdghosdjgpsjgrojjoeejgGGGGGGGGsdgsesgeeGGGasfgegegesgggggggsdasgsdgrgrE-UWR-")
</script>

<template>
  <div style="width: 300px">
    <div>
      {{ type }}
    </div>
      <el-input :model-value='value' readonly type="textarea" :autosize="{ minRows: 2, maxRows: 6 }"></el-input>
  </div>
</template>
```

### What is Expected?

the 2nd `el-input` height can auto change like the 1st `el-input`

### What is actually happening?

the 2nd `el-input` height doesn't auto change

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by MrWeilian**
fix: #12074 

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/input/src/input.vue [hunk 1]
```diff
@@ -312,7 +312,10 @@ const suffixVisible = computed(
 
 const [recordCursor, setCursor] = useCursor(input)
 
+const isElHidden = textarea.value?.offsetParent === null
+
 useResizeObserver(textarea, (entries) => {
+  !isElHidden && resizeTextarea()
```
Comment (chenxch) [inline packages/components/input/src/input.vue:318]: This causes resize to fire repeatedly.

[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IExpc3QgZnJvbSBcIi4vTGlzdC52dWVcIlxuXG5jb25zdCBzaG93VHlwZSA9IHJlZignMScpXG5jb25zdCBvbkNoYW5nZSA9ICgpID0+IHtcbiAgc2hvd1R5cGUudmFsdWUgPSBzaG93VHlwZS52YWx1ZSA9PT0gXCIxXCIgPyBcIjJcIiA6IFwiMVwiXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2PlxuICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBAY2xpY2s9XCJvbkNoYW5nZVwiPmNoYW5nZTwvZWwtYnV0dG9uPlxuICAgIDxMaXN0IHYtc2hvdz1cInNob3dUeXBlID09PSAnMSdcIiB0eXBlPVwiMVwiIGtleT1cIjFcIj48L0xpc3Q+XG4gICAgPExpc3Qgdi1zaG93PVwic2hvd1R5cGUgPT09ICcyJ1wiIHR5cGU9XCIyXCIga2V5PVwiMlwiPjwvTGlzdD5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTEyMDg3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTIwODctZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctMTIwODctZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiTGlzdC52dWUiOiI8c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmLCB2ZXJzaW9uIGFzIHZ1ZVZlcnNpb24gfSBmcm9tICd2dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIEVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG4gIFxuZGVmaW5lUHJvcHM8e1xuICB0eXBlOiBzdHJpbmdcbn0+KClcblxuY29uc3QgdmFsdWUgPSByZWYoXCJhSE9JRlZTREdHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dzZGdob3Nkamdwc2pncm9qam9lZWpnR0dHR0dHR0dzZGdzZXNnZWVHR0dhc2ZnZWdlZ2VzZ2dnZ2dnZ3NkYXNnc2RncmdyRS1VV1ItXCIpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2IHN0eWxlPVwid2lkdGg6IDMwMHB4XCI+XG4gICAgPGRpdj5cbiAgICAgIHt7IHR5cGUgfX1cbiAgICA8L2Rpdj5cbiAgICAgIDxlbC1pbnB1dCA6bW9kZWwtdmFsdWU9J3ZhbHVlJyByZWFkb25seSB0eXBlPVwidGV4dGFyZWFcIiA6YXV0b3NpemU9XCJ7IG1pblJvd3M6IDIsIG1heFJvd3M6IDYgfVwiPjwvZWwtaW5wdXQ+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbiIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTEyMDg3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

adjust textarea height manually

Comment (MrWeilian) [reply packages/components/input/src/input.vue]: resolved~

File: packages/components/input/src/input.vue [hunk 2]
```diff
@@ -312,7 +312,10 @@ const suffixVisible = computed(
 
 const [recordCursor, setCursor] = useCursor(input)
 
+const isElHidden = () => textarea.value?.offsetParent === null
+
 useResizeObserver(textarea, (entries) => {
+  !isElHidden() && onceInitSizeTextarea()
```
Comment (HeftyKoo) [inline packages/components/input/src/input.vue:318]: May be better to check `autosize` first to avoid useless reflow.

Comment (MrWeilian) [reply packages/components/input/src/input.vue]: resolved~

Comment (HeftyKoo) [reply packages/components/input/src/input.vue]: LGTM

**Review by HeftyKoo — APPROVED** [review]

**Review by chenxch — APPROVED** [review]

**Pull request merged**